gopkgs.go

```diff
@@ -358,25 +378,75 @@ func List(opts Options) (map[string]Pkg, error) {
 }
 
 type mod struct {
-	path string
-	dir  string
+	path      string
+	pkgDir    string
+	vendorDir string
 }
 
-func listMods(workDir string) ([]mod, error) {
-	cmdArgs := []string{"list", "-m", "-f={{.Path}};{{.Dir}}", "all"}
-	cmd := exec.Command("go", cmdArgs...)
-	cmd.Dir = workDir
-	out, err := cmd.Output()
+func listMods(workDir string, vendorMode bool) ([]mod, error) {
+	// exec `go list -m ...` to get module path list
+	// if GOFLAGS contains '-mod=vendor', it gets vendor path list instead
+	s, err := execGoList(workDir, "list", "-m", "-f={{.Path}};{{.Dir}}", "all")
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "unable to execute `go list -m ...`")
 	}
 
 	var mods []mod
-	s := bufio.NewScanner(bytes.NewReader(out))
 	for s.Scan() {
 		line := s.Text()
 		ls := strings.Split(line, ";")
-		mods = append(mods, mod{path: ls[0], dir: ls[1]})
+		if vendorMode {
+			mods = append(mods, mod{path: ls[0], vendorDir: ls[1]})
+		} else {
+			mods = append(mods, mod{path: ls[0], pkgDir: ls[1]})
+		}
 	}
+
+	// if GOFLAGS contains '-mod=vendor', exec `go list -mod= -m ...` to fill the module paths as well
+	if vendorMode {
+		s, err = execGoList(workDir, "list", modEmptyFlag, "-m", "-f={{.Path}};{{.Dir}}", "all")
```

Comment:
What if we always use `-mod=""`? is there any other mod outside the `-mod=vendor`? Means we ignoring `-mod=...` from `GOFLAGS`.

Reply:
That seemed to do the trick, although keep in mind that with this we'll ignore `/vendor` completely.